File: TestPilot/Mocks/Generation/Swift/BankAccountManager.swift
import Foundation

struct BankAccount {
    let id: Int
    let owner: String
    private(set) var balance: Double
}

enum BankAccountError: Error {
    case insufficientFunds
    case accountNotFound
}

class BankAccountManager {
    private var accounts: [BankAccount] = []

    func createAccount(_ account: BankAccount) {
        accounts.append(account)
    }

    func deposit(to accountId: Int, amount: Double) throws {
        guard let index = accounts.firstIndex(where: { $0.id == accountId }) else {
            throw BankAccountError.accountNotFound
        }
        accounts[index] = BankAccount(
            id: accounts[index].id,
            owner: accounts[index].owner,
            balance: accounts[index].balance + amount
        )
    }

    func withdraw(from accountId: Int, amount: Double) throws {
        guard let index = accounts.firstIndex(where: { $0.id == accountId }) else {
            throw BankAccountError.accountNotFound
        }
        let currentBalance = accounts[index].balance
        guard currentBalance >= amount else {
            throw BankAccountError.insufficientFunds
        }
        accounts[index] = BankAccount(
            id: accounts[index].id,
            owner: accounts[index].owner,
            balance: currentBalance - amount
        )
    }

    func balance(for accountId: Int) throws -> Double {
        guard let account = accounts.first(where: { $0.id == accountId }) else {
            throw BankAccountError.accountNotFound
        }
        return account.balance
    }
}

File: Mocks/Generation/Python/bank_account_manager.py
class BankAccountError(Exception):
    pass

class InsufficientFundsError(BankAccountError):
    pass

class AccountNotFoundError(BankAccountError):
    pass

class BankAccount:
    def __init__(self, account_id, owner, balance=0.0):
        self.id = account_id
        self.owner = owner
        self.balance = balance

class BankAccountManager:
    def __init__(self):
        self._accounts = {}

    def create_account(self, account):
        self._accounts[account.id] = account

    def deposit(self, account_id, amount):
        account = self._get_account(account_id)
        account.balance += amount

    def withdraw(self, account_id, amount):
        account = self._get_account(account_id)
        if account.balance < amount:
            raise InsufficientFundsError("Insufficient funds.")
        account.balance -= amount

    def get_balance(self, account_id):
        account = self._get_account(account_id)
        return account.balance

    def _get_account(self, account_id):
        if account_id not in self._accounts:
            raise AccountNotFoundError(f"Account {account_id} not found.")
        return self._accounts[account_id]
